src/mame/drivers/jungleyo.cpp

```diff
@@ -178,21 +293,23 @@ void jungleyo_state::main_map(address_map &map)
 	map(0xa0032a, 0xa0032b).portr("DSW12");
 	map(0xa0032c, 0xa0032d).portr("DSW34");
 	map(0xa00336, 0xa00337).w(FUNC(jungleyo_state::output_w)); // outputs? observed values: lots
-	map(0xa00689, 0xa00689).rw("oki", FUNC(okim6295_device::read), FUNC(okim6295_device::write)); // TODO: something's definitely wrong here, lots of errors in the log
+	map(0xa00689, 0xa00689).rw("oki", FUNC(okim6295_device::read), FUNC(okim6295_device::write));
 	//map(0xa0068a, 0xa0068b).nopw(); // observed values: 0x0101
 	//map(0xa0082e, 0xa0082f).nopw(); // observed values: 0x0000, 0x4000, 0x5900 (double up),
-	//map(0xa00830, 0xa00831).nopw(); // observed values: 0x0000 and 0x0007
-	//map(0xa00832, 0xa00833).nopw(); // tilemap priority ? observed values: 0x0000 and 0x0001
-	//map(0xa00aaa, 0xa00aab).nopw(); // observed values: 0x0000, this is written continuously
+	map(0xa00831, 0xa00831).w(FUNC(jungleyo_state::layer_enable_w));
+	map(0xa00833, 0xa00833).w(FUNC(jungleyo_state::video_priority_w));
+	map(0xa00aaa, 0xa00aab).nopw(); // irq ack or watchdog (written at start of vblank irq)
 	//map(0xa00d58, 0xa00d59).nopw(); // observed values: 0x0004
 	//map(0xa00e9a, 0xa00e9b).nopw(); // observed values: 0x0005
 	//map(0xa00fc6, 0xa00fc7).nopw(); // observed values: 0x0006
-	map(0xb00000, 0xb0ffff).ram();
-	map(0xb10000, 0xb1ffff).ram();
-	map(0xb20000, 0xb2ffff).ram();
+	map(0xb00000, 0xb2ffff).rw(FUNC(jungleyo_state::palette_ram_r), FUNC(jungleyo_state::palette_ram_w)).umask16(0x00ff);
 	map(0xb80000, 0xb81fff).ram().share(m_fg_videoram).w(FUNC(jungleyo_state::fg_videoram_w));
 	map(0xb90000, 0xb91fff).ram().share(m_bg_videoram).w(FUNC(jungleyo_state::bg_videoram_w));
-	map(0xba0000, 0xba1fff).ram();
+	map(0xba0000, 0xba07ff).ram().share(m_reel_vram[0]).w(FUNC(jungleyo_state::reel_vram_w<0>));
+	map(0xba0800, 0xba0fff).ram().share(m_reel_vram[1]).w(FUNC(jungleyo_state::reel_vram_w<1>));
+	map(0xba1000, 0xba17ff).ram().share(m_reel_vram[2]).w(FUNC(jungleyo_state::reel_vram_w<2>));
+	map(0xba1800, 0xba1fff).ram(); // supposedly the 4th reel VRAM bank, inited at POST only and prolly just tied to NOP instead
```

Comment:
Is it really too much to spell “probably” correctly?

Reply:
Excuse me?

![cuavas_prolly](https://user-images.githubusercontent.com/9251271/105999099-988e3f80-60ad-11eb-898d-03e4589b82d9.png)

Reply:
it is very informal (and I can't remember the last time anybody I know here used it)

Reply:
I used to use it a lot for online games back in the 80's/90's/00's .. but have not used it all that much in recent times.

Reply:
It’s not a real word, it’s a phonetic spelling of a lazy mispronunciation.  We have higher standards than that.

Reply:
7 hits in the source code, and this is the first time I hear that "prolly" is against "standards" (?).

Reply:
The 6 cases in the MAME code, organised chronologically:
28 Jun 2017: https://github.com/mamedev/mame/commit/12824956e936e9716a7d29c9dd18f12aeec9632d#diff-b98cf4a01eb48be4ccf23fdefa7c84f2a8aa300e595e3ea4abf3a7e7cf85050aR40
15 Feb 2018: https://github.com/mamedev/mame/commit/852a6533e9f2afd3f4d7589a7f254b782a54b610#diff-56bb0d2e3fba56f29dfe5497594d12ed60a3d56fddb6047f5726e84c830c8dffR94
24 May 2019: https://github.com/mamedev/mame/commit/9af3931f194613761eae93a20d7bf8495031e2f4#diff-7fff0e3c1abc3bed817c80ae0909b4cff0623eaddd7842d0f09e8b09805ad2bfR518
24 Jul 2019: https://github.com/mamedev/mame/commit/e721e1f4ff4572876bebee1da56d8b6f27745e2c#diff-a9f92a3049a8a9c19ad7a5b99ecadba8a4d38864ecea1a249dcdc2c97cddc50fR538
24 Aug 2019: https://github.com/mamedev/mame/commit/f5b2fd9ee6ef6ced1eec7859494e82c375d9c8d2#diff-148eb445a116bce4716f4f53f037d4453755ab463cb8f32830057df3d73d3197R243
13 Dec 2019: https://github.com/mamedev/mame/commit/f58c3ea2644c51559a45d09fa1c3a146410bcf5a#diff-64372177e8768593bf91f048ae7ec7e493858326cbf183269f60b361e641faaaR734